src/main/java/uk/gov/register/db/RecordQueryDAO.java

```diff
@@ -21,26 +21,25 @@
 import java.util.Optional;
 
 public abstract class RecordQueryDAO {
-
     private static final ObjectMapper objectMapper = Jackson.newObjectMapper();
 
     @SqlQuery("SELECT count FROM total_records")
     public abstract int getTotalRecords();
 
-    @SqlQuery("select entry_number, timestamp, e.sha256hex as sha256hex, key, content from entry e, item i where e.sha256hex=i.sha256hex and e.entry_number = (select entry_number from current_keys where current_keys.key=:key)")
+    @SqlQuery("select e.entry_number, array_agg(ei.sha256hex) as sha256hex, e.timestamp, e.key, array_agg(i.content) as content from entry e join entry_item ei on ei.entry_number = e.entry_number and e.entry_number = (select entry_number from current_keys where current_keys.key=:key) join item i on i.sha256hex = ei.sha256hex group by e.entry_number")
     @SingleValueResult(Record.class)
     @RegisterMapper(RecordMapper.class)
     public abstract Optional<Record> findByPrimaryKey(@Bind("key") String key);
 
-    @SqlQuery("select entry.entry_number, timestamp, entry.sha256hex as sha256hex, entry.key, content from item, entry, current_keys where current_keys.entry_number = entry.entry_number and item.sha256hex=entry.sha256hex order by entry.entry_number desc limit :limit offset :offset")
+    @SqlQuery("select e.entry_number, array_agg(ei.sha256hex) as sha256hex, e.timestamp, e.key, array_agg(i.content) as content from entry e join entry_item ei on ei.entry_number = e.entry_number join item i on i.sha256hex = ei.sha256hex join current_keys ck on ck.entry_number = e.entry_number group by e.entry_number order by e.entry_number desc")
     @RegisterMapper(RecordMapper.class)
     public abstract List<Record> getRecords(@Bind("limit") long limit, @Bind("offset") long offset);
 
-    @SqlQuery("select entry_number, timestamp, sha256hex, key from entry where sha256hex in (select sha256hex from item where (content @> :contentPGobject)) order by entry_number asc")
+    @SqlQuery("select e.entry_number, array_agg(ei.sha256hex) as sha256hex, e.timestamp, e.key from entry e join entry_item ei on ei.entry_number = e.entry_number join item i on i.sha256hex = ei.sha256hex and i.content @> :contentPGobject group by e.entry_number order by e.entry_number asc")
```

Comment:
This query can be made much simpler now that we have `key` in the `entry` table - we didn't used to have this, which is why we used the jsonb query. Instead we could do:

```
select e.entry_number, array_remove(array_agg(ei.sha256hex), null) as sha256hex, e.timestamp, e.key from entry e left join entry_item ei on ei.entry_number = e.entry_number where e.key = :key group by e.entry_number;
```